docs/guides/intro-to-tests.md

```diff
@@ -0,0 +1,154 @@
+
+```js
+{ "hello" : 1 }
+```
+
+Remember that `createWordCount` needs to return an object with a count of each word in the string. In the case where the input is `"hello"` we only have a single word string and therefore a count of 1 for `"hello"`.
+
+### Case 2 🖊️
+
+```js
+const input = "north by north west";
+const result = createWordCount(input);
+```
+
+In example 2, `input` stores a string `"north by north west"` which we pass as an argument to `createWordCount`.
+We then store the return value of `createWordCount` in a variable called `output`.
+In this case, we would expect the `output` to be:
+
+```js
+{
+    "north" : 2,
+    "by" : 1,
+    "west" : 1
+}
+```
+
+## 🧪 What is a test ?
+
+We can now define what a test is in software engineering.
+
+> 🔑 A **test** is any piece of code that **checks a certain unit of code** ( typically a function ) **behaves in a particular way**.
+
+For example, we could write a test to check that `createWordCount` behaves in a particular way when passed a string containing a single word. For this example, we're using a popular JavaScript testing framework called [Jest](https://jestjs.io/docs/getting-started) to write the test code. We'll examine the example below written using Jest, and then break down the different parts of the test code to understand how it all works.
+
+```js
+test("will count the number of words for a single word string", function () {
+  const input = "hello";
+  const actualResult = createWordCount(input);
+  const expectedResult = {
+    hello: 1,
+  };
+  expect(actualResult).toEqual(expectedResult);
+});
+```
+
+Let's break down the different parts of this test code.
+
+## Anatomy of a test
+
+#### 1. Test description
+
+```js
+test("will count the number of words in a single word string", function () {});
+```
+
+We use a function in Jest called `test` to write a test description - `"will count the number of words for a single word string"` which describes the behaviour we expect in this particular test. There is also a function `function () {}` in which we will write our test code.
+
+#### 2. Test setup
+
+```js
+const input = "hello";
+const actualResult = createWordCount(input);
+const expectedResult = {
+  hello: 1,
+};
+```
+
+In this section, we're declaring some important values that we will need to compare later on in the test.
+We declare an `input` variable that stores a value that we will pass to the function we are testing - `createWordCount`.
+We declare an `actualResult` variable that stores the **return value** of our function `createWordCount`.
+We declare an `expectedResult` which stores what we **expect** the `actualResult` to be when our function is behaving according to the test description.
+
+❗ Note: The test setup will vary from test to test - so you can expect to see different variable names altogether in different tests. The setup here is done in order to make it clearer what values will be compared later on in the test.
```

Comment:
Fwiw, if you really want to call attention to something, we can use Docusaurus' "admonitions" feature: https://docusaurus.io/docs/markdown-features/admonitions

Reply:
Ohh nice didn't know you could do that